### PR TITLE
Feat: Add create new organization - 161

### LINKED ISSUE #161: LCFS - IDIR users can create new Organizations
**Description:**
IDIR users can create new organizations. 

- Add Create organization button on the IDIR Organizations page
- Button is only visible to IDIR users with the Administrator role
- Develop the 'create organization' page, see wireframe
- make certain fields required
- Save and back buttons
- Clicking on the Save button adds the record to the database and the entry appears in the react table on the Organizations page; it also takes the user back to the Organizations page and the user can see the organization in the Organizations react table
- Back button takes the user back to the Organizations page
- Includes "same as..." check boxes that allow the user to indicate that their legal and operating names are the same, and that their address for service and record keeping address are the same. 
- Telephone number is free form text to allow for users to input extensions

**Organization information: required fields**
- Legal name of organization
- Operating name of organization
- Telephone number
- Address for service (postal address)
- Email address
- Address in B.C. at which records are maintained
- Supplier type

**Organization information: optional fields**
- Company profile, EDRMS record #

**Wireframe:** 
- New organization button: https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164769377
- Create org page: https://preview.uxpin.com/59145260c7d8cc48969ac79401cc03d84843c3d9#/pages/164728479/simulate/no-panels

**Purpose and benefit to user:**
Allows IDIR user to create new organizations

**Acceptance Criteria:**

- [ ] Given I am a IDIR user, When I am clicking Organization details, then  should be able to create an organization.

**Development Checklist:**
- [ ] Test coverage both on UI and backend.

**Notes:**

### PR DESCRIPTION
**Pull request by hamed-valiollahi**
This PR introduces the feature to create new organization.

- Please note that the migration needs to be executed.
- Additionally, three new frontend packages have been added.

Closes #161

### CODE REVIEW

File: backend/lcfs/web/api/organization/views.py [hunk 1]
```diff
@@ -1,33 +1,88 @@
...
+        # Create and add organization model to the database
+        org_model = Organization(**org_data_dict)
+        db.add(org_model)
+        await db.commit()
+        await db.refresh(org_model)
+
+        return org_model
+
+    except SQLAlchemyError as sql_ex:
+        await db.rollback()
```
Comment (AlexZorkin) [inline backend/lcfs/web/api/organization/views.py:69]: Given that we initially create the 'org_address' and 'org_attorney_address', followed by the 'org_model', does the rollback mechanism effectively address scenarios where 'org_address' is successfully created, but an error occurs during the commit of the 'org_model'?

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -120,7 +121,15 @@ const router = createBrowserRouter([
       {
         path: appRoutes.EDIT_USER,
         element: <EditUser />
-      }
+      },
+      // TODO: Modify OrganizationLayout to support child route integration, then use it for the following route.
```
Comment (AlexZorkin) [inline frontend/src/App.jsx:125]: Routes on our release-0.2.0 branch have been updated to account for this. Please either merge or rebase on top of this branch to get latest changes.

File: frontend/src/components/Loading.jsx [hunk 3]
```diff
@@ -1,3 +1,32 @@
-const Loading = (props) => <div className="text-center">Loading...</div>
+import PropTypes from 'prop-types'
```
Comment (AlexZorkin) [inline frontend/src/components/Loading.jsx:1]: There will be a merge conflict with this loading component, but I like your approach so include please it.

File: frontend/src/constants/routes.js [hunk 4]
```diff
@@ -6,8 +6,11 @@ export const ORGANIZATION = '/organization'
 export const ORGANIZATION_USERS = `${ORGANIZATION}/users`
 export const ORGANIZATION_USER = `${ORGANIZATION_USERS}/:userID`
 
-export const ADMINISTRATION = '/administration'
-export const ADMINISTRATION_USERS = `${ADMINISTRATION}/users`
+// TODO: Use 'organizations' in plural form for consistency with other routes.
```
Comment (AlexZorkin) [inline frontend/src/constants/routes.js:9]: yes, we will rework the routing conventions next sprint.

File: frontend/src/views/organizations/AddOrganization.jsx [hunk 5]
```diff
@@ -0,0 +1,745 @@
...
+import { useMutation } from 'react-query'
+
+// Internal Modules
+import BCAlert from '@/components/BCAlert'
+import Loading from '@/components/Loading'
+import useApiService from '@/services/useApiService'
+import * as routes from '@/constants/routes'
+
+// Schema for form validation
+const schema = Yup.object({
```
Comment (AlexZorkin) [inline frontend/src/views/organizations/AddOrganization.jsx:33]: Nice validation!

**Review by AlexZorkin — CHANGES_REQUESTED** [review]
Excellent work. Lots of new conventions we can utilize moving forward. Just a couple of comments to address.

File: backend/lcfs/web/api/organization/views.py (hunk 1 above)
Comment (hamed-valiollahi) [reply backend/lcfs/web/api/organization/views.py:69]: Thanks, Alex, for pointing that out. I enhanced the code to better handle transactions. Now, if 'org_address' is created but 'org_model' fails, the rollback will undo the 'org_address' creation.

File: frontend/src/App.jsx (hunk 2 above)
Comment (hamed-valiollahi) [reply frontend/src/App.jsx:125]: Sure! Thanks for the information.

File: frontend/src/components/Loading.jsx (hunk 3 above)
Comment (hamed-valiollahi) [reply frontend/src/components/Loading.jsx:1]: Sure, I'll include this loading component. Thanks!

File: frontend/src/views/organizations/AddOrganization.jsx (hunk 5 above)
Comment (hamed-valiollahi) [reply frontend/src/views/organizations/AddOrganization.jsx:33]: Thanks, Alex!

**Comment by hamed-valiollahi** [comment]
> Excellent work. Lots of new conventions we can utilize moving forward. Just a couple of comments to address.

Thanks a lot, Alex, for reviewing my code!

**Review by AlexZorkin — APPROVED** [review]

**Pull request merged**